flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -740,20 +740,19 @@ public WebMonitorEndpoint(
                         jobVertexBackPressureHandler.getMessageHeaders(),
                         jobVertexBackPressureHandler));
 
-        if (clusterConfiguration.get(RestOptions.ENABLE_FLAMEGRAPH)) {
-            final JobVertexFlameGraphHandler jobVertexFlameGraphHandler =
-                    new JobVertexFlameGraphHandler(
-                            leaderRetriever,
-                            timeout,
-                            responseHeaders,
-                            executionGraphCache,
-                            executor,
-                            initializeThreadInfoTracker(executor));
-            handlers.add(
-                    Tuple2.of(
-                            jobVertexFlameGraphHandler.getMessageHeaders(),
-                            jobVertexFlameGraphHandler));
-        }
+        final JobVertexFlameGraphHandler jobVertexFlameGraphHandler =
+                new JobVertexFlameGraphHandler(
+                        leaderRetriever,
+                        timeout,
+                        responseHeaders,
+                        executionGraphCache,
+                        executor,
+                        initializeThreadInfoTracker(executor),
```

Comment:
Is it actually worth to also skip creating this if Flamegraphs are not enabled? How much overhead is this?

Reply:
Sorry mate, I neglected the unnecessary workload of executing ’initializeThreadInfoTracker‘. By enabling it Flink will keep capturing Thead dump but we don't use it at all. I will fix it.